### ISSUE TITLE
🐞 Dagger init can't run on windows

### ISSUE THREAD

**Issue by kerlos** [dagger/dagger#6923]
### What is the issue?

Cannot run `dagger init` on windows

seems it has problem with `filepath.IsAbs` for windows

### Dagger version

dagger v0.10.2 windows/amd64

### Steps to reproduce

1. install dagger CLI for windows
2. run command `dagger init`

### Log output

from cli

```
Error: failed to get configured module: failed to get local root path: input: moduleSource.resolveContextPathFromCaller resolve: failed to find up root: path is not absolute: C:\projects\test
```

from dagger engine

```
2024-03-25 16:24:26 time="2024-03-25T09:24:26Z" level=warning msg="error resolving moduleSource(refString: \".\").resolveContextPathFromCaller: String!" error="map[error:failed to find up root: path is not absolute: C:\\projects\\test kind:*fmt.wrapError stack:<nil>]" source="&{github.com/dagger/dagger/tracing.AroundFunc.ProgrockAroundFunc.func1 /app/tracing/graphql.go 107}"
```

**Comment by jedevc** [comment]
:wave: aha, we saw this last week at kubecon - there's an internal issue somewhere (cc @jpadams fyi)

TL;DR - the issue is here, https://github.com/dagger/dagger/blob/288c7f7e5d9e03e6de586934b0e99962b0fba131/engine/client/client.go#L716-L721

I think we need a call to `filepath.ToSlash`, and probably need to make sure fsutil/buildkit is up-to-date for good measure as well. But also, I don't have a current windows machine to play on, so we should also make sure we have some CI/etc as well.

**Comment by kerlos** [comment]
I look into source of `filepath.IsAbs` it should handle both `\\` and `/` 
https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/path/filepath/path_windows.go;l=59

In this playground if it run on windows it both return true but if run on playground website. both will be false
https://go.dev/play/p/3ROFwTgLDZT

**Comment by kerlos** [comment]
I build dagger cmd seems its a problem of core package. i think core package run with linux so it doesn't understand windows path

https://[github.com/dagger/dagger/blob/main/core/schema/modulesource.go#L1056](https://github.com/dagger/dagger/blob/main/core/schema/modulesource.go#L1056)
